### PR TITLE
Introduce Prometheus monitoring example

### PR DESCRIPTION
**Pull request by elanv**
This PR includes Prometheus monitoring example and updated user doc.
extraPorts and podMonitor CR of Prometheus operator is used for the example.

### CODE REVIEW

**Comment by elanv** [comment]
Related issue: #243

**Comment by functicons** [comment]
Thanks! Will follow the doc to try and give you feedback later today.

File: docs/user_guide.md [hunk 1]
```diff
@@ -378,6 +378,22 @@ Status:
     Update Time:     2020-04-03T10:04:50+09:00
 ```
 
+### Monitoring with Prometheus
+
+Flink cluster can be monitored with Prometheus in various ways. Here, we introduce the method using podMonitor
```
Comment (functicons) [inline docs/user_guide.md:383]: s/podMonitor/PodMonitor, also update other places.

File: docs/user_guide.md [hunk 2]
```diff
@@ -378,6 +378,22 @@ Status:
...
+custom resource of Prometheus operator. First, create a Flink cluster the metric exporter activated and its port exposed.
```
Comment (functicons) [inline docs/user_guide.md:384]: Add a link to Prometheus operator.

Comment (functicons) [inline docs/user_guide.md:384]: Missing a "with" between "Flink cluster" and "the metric exporter"?

File: docs/user_guide.md [hunk 3]
```diff
@@ -378,6 +378,22 @@ Status:
...
+Next, create a podMonitor which will be used to generate service discovery configurations and register it to Prometheus.
+Exposed Flink metric port must be set as the endpoint of the podMonitor. See the
+[Prometheus API docs](https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md) for details.
+
+You can create Prometheus metric exporter activated [Flink cluster](../examples/prometheus/flink_metric_cluster.yaml) and
+[pod monitor](../examples/prometheus/pod-monitor.yaml) like this.
+
+```bash
+kubectl apply -f examples/prometheus/flink_metric_cluster.yaml
+kubectl apply -f examples/prometheus/pod-monitor.yaml
```
Comment (functicons) [inline docs/user_guide.md:394]: Introduce how to open the web UI after these commands.

File: docs/user_guide.md (hunk 1 above)
Comment (elanv) [reply docs/user_guide.md]: Done.

File: docs/user_guide.md (hunk 2 above)
Comment (elanv) [reply docs/user_guide.md]: Done.

Comment (elanv) [reply docs/user_guide.md]: Done.

File: docs/user_guide.md (hunk 3 above)
Comment (elanv) [reply docs/user_guide.md:395]: Done.

**Comment by yalctay93** [comment]
I've tried this solutions however, extraPorts are not created when deploying the flinkcluster. Metrics are available when using curl localhost:9249 but my pod-monitor apparently is not scraping it.

**Pull request merged**